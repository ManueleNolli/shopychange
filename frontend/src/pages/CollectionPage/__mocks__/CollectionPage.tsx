import { divWithoutChildrenMock } from '../../../../mocksUtils'

const CollectionPageMock = () => divWithoutChildrenMock('CollectionPage')
export default CollectionPageMock
