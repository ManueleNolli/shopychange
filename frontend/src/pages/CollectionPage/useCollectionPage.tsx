import { useEffect, useState } from 'react'
import {
  collectionQuery,
  handlerQueryErrorCatch,
} from '../../services/BackendService'
import useFetching from '../../hooks/useFetching'
import { useNavigate } from 'react-router-dom'
import { useUserContext } from '../../context/userContext'
import { Collection } from '../../types/components/Collection'

export default function useCollectionPage(contractAddress: string) {
  const navigate = useNavigate()
  const { blockchainNetworkId } = useUserContext()
  const { isFetching, setIsFetching, isError, setIsError } = useFetching()
  const [collection, setCollection] = useState<Collection>({
    contract: {
      name: '',
      symbol: '',
      address: '',
    },
    nfts: [],
  })

  useEffect(() => {
    const fetchCollection = async () => {
      if (!blockchainNetworkId) return

      const response = await handlerQueryErrorCatch(
        collectionQuery(blockchainNetworkId, contractAddress)
      )

      if (response.error) {
        setIsError(true)
        setIsFetching(false)
        return
      }

      const collection = response.result.collection as Collection
      setCollection(collection)
      setIsFetching(false)
    }

    fetchCollection()
  })

  const onNavigate = () => {
    navigate(`/${contractAddress}/royalties`)
  }

  return {
    isFetching,
    isError,
    collection,
    onNavigate,
  }
}
