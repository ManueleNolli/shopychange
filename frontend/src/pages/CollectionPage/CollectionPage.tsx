import React from 'react'

import { Box, Center, Heading, Flex } from '@chakra-ui/react'
import NFTCardGrid from '../../components/NFTCardGrid/NFTCardGrid'
import { CollectionPageProps } from '../../types/props/CollectionPageProps'
import CollectionOwnerCheck from '../../utils/CollectionOwnerCheck/CollectionOwnerCheck'
import ShadowButton from '../../components/ShadowButton/ShadowButton'
import { colors } from '../../styles/Colors'
import WithdrawRoyaltyButton from '../../components/WithdrawRoyaltyButton/WithdrawRoyaltyButton'
import useCollectionPage from './useCollectionPage'
import QueryContainer from '../../utils/QueryContainer/QueryContainer'

export default function CollectionPage({
  contractAddress,
}: CollectionPageProps) {
  const { isFetching, isError, collection, onNavigate } =
    useCollectionPage(contractAddress)

  return (
    <Box>
      <QueryContainer isLoading={isFetching} isError={isError}>
        <Center>
          <Flex direction="column">
            <Heading textAlign={'center'}>{collection.contract.name}</Heading>
            <Center my="2vh">
              <CollectionOwnerCheck
                collectionAddress={contractAddress}
                redirectHome={false}
              >
                <ShadowButton
                  label={'Edit Collection Royalties'}
                  bgColor={`${colors.primaryChakraColor}.400`}
                  textColor={'white'}
                  shadowColor={colors.primaryShadowColor}
                  hoverColor={`${colors.primaryChakraColor}.500`}
                  focusColor={`${colors.primaryChakraColor}.500`}
                  buttonProps={{
                    w: { base: '60vw', md: '40vw', lg: '20vw' },
                    maxW: '20em',
                    mr: '2vw',
                  }}
                  onClick={onNavigate}
                />
              </CollectionOwnerCheck>
              <WithdrawRoyaltyButton collectionAddress={contractAddress} />
            </Center>

            <NFTCardGrid nfts={collection.nfts} />
          </Flex>
        </Center>
      </QueryContainer>
    </Box>
  )
}
