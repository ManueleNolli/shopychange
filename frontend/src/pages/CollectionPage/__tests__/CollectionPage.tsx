import React from 'react'
import { render } from '@testing-library/react'
import CollectionPage from '../CollectionPage'
import useCollectionPage from '../useCollectionPage'

jest.mock('../../../components/NFTCardGrid/NFTCardGrid')
jest.mock('../../../components/WithdrawRoyaltyButton/WithdrawRoyaltyButton')
jest.mock('../useCollectionPage', () => jest.fn())
jest.mock('../../../utils/CollectionOwnerCheck/CollectionOwnerCheck', () => ({
  __esModule: true,
  default: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="CollectionOwnerCheck">{children}</div>
  ),
}))
const mockProps = {
  contractAddress: '0x1203A6e8bD6A063AEa9D4B7F8A2BaE9405fE44Cd',
}

const mockReturnValue = {
  collection: {
    contract: {
      name: 'mockedContractName',
      symbol: 'mockedContractSymbol',
      address: 'mockedContractAddress',
    },
    nfts: [
      {
        contractAddress: 'mockedContractAddress',
        tokenId: 0,
        image: 'mockedImage',
        name: 'mockedName',
      },
      {
        contractAddress: 'mockedContractAddress1',
        tokenId: 1,
        image: 'mockedImage1',
        name: 'mockedName1',
      },
    ],
  },
}

describe('CollectionPage', () => {
  test('renders successfully', () => {
    ;(useCollectionPage as jest.Mock).mockReturnValue({
      isFetching: false,
      isError: false,
      collection: mockReturnValue.collection,
      onNavigate: jest.fn(),
    })

    const tree = render(<CollectionPage {...mockProps} />)
    expect(tree.container).toMatchSnapshot()
  })
})
