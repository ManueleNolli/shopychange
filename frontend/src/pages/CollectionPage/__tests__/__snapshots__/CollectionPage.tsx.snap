// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage renders successfully 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-j7qwjs"
      >
        <h2
          class="chakra-heading css-xi606m"
        >
          mockedContractName
        </h2>
        <div
          class="css-1he8q8i"
        >
          <div
            data-testid="CollectionOwnerCheck"
          >
            <button
              class="chakra-button css-h1c5vj"
              data-testid="onClick-button"
              style="white-space: initial;"
              type="button"
            >
              Edit Collection Royalties
            </button>
          </div>
          <div
            data-testid="WithdrawRoyaltyButton"
          />
        </div>
        <div
          data-testid="NFTCardGrid"
        />
      </div>
    </div>
  </div>
</div>
`;
